### PR TITLE
Update URL Parsing

### PR DESCRIPTION
**Pull request by MarkAckert**
Fixes https://github.com/zowe/vscode-extension-for-zowe/issues/307

### CODE REVIEW

**Comment by MarkAckert** [comment]
In addition to URL parsing fixes, I've changed URL placeholder text from `http(s)` to `https`. I confirmed that z/OSMF Profiles for the CLI only support HTTPS, so `http(s)` would be confusing to end-users who expect an `http` URL to work. `validateUrl` will now only accept URLs with the `https` protocol.

**Comment by jellypuno** [comment]
@IgorCATech I think we need to change/add something the documentation for the URL. 
- We are only accepting https://
- If port is not specified, the defaut value will be 443.

File: src/Profiles.ts [hunk 1]
```diff
@@ -104,15 +105,40 @@ export class Profiles { // Processing stops if there are no profiles detected
         return this.allProfiles;
     }
 
-    public validateUrl = (newUrl: string) => {
+    public validateAndParseUrl = (newUrl: string): IUrlValidator => {
+
+        let url: URL;
+        const validProtocols: string[] = ["https"];
+        const DEFAULT_HTTPS_PORT: number = 443;
```
Comment (zFernand0) [inline src/Profiles.ts]: No need to implement this now, but what if someone wanted to have a different default port?
Should this be a setting in our set of vscode configurable items?

File: src/Profiles.ts [hunk 2]
```diff
@@ -104,15 +105,40 @@ export class Profiles { // Processing stops if there are no profiles detected
...
-        validURL = url.hostname;
-        validPort = Number(url.port);
-        return url.port ? true : false;
+
+        // if port is empty, set https defaults
+        if (!url.port.trim()) {
+            validationResult.port = DEFAULT_HTTPS_PORT;
+        }
+        else {
+            validationResult.port = Number(url.port);
```
Comment (zFernand0) [inline src/Profiles.ts]: Will this be a valid url `https://10.142.0.23:65536`?

Comment (jellypuno) [reply src/Profiles.ts]: > Will this be a valid url `https://10.142.0.23:65536`?

I tested this one and it is not considered a valid URL

File: src/Profiles.ts (hunk 1 above)
Comment (zFernand0) [reply src/Profiles.ts]: Should we allow the default port to be handled by nodejs depending on the protocol?

**Comment by zFernand0** [comment]
@MarkAckert, could you resolve the conflicts?

**Comment by zFernand0** [comment]
:+1:

**Pull request merged**